src/service/navigation.js

```diff
@@ -409,6 +419,45 @@ export class Navigation {
     }
   }
 
+  /**
+   * @param {!Function} callback
+   * @param {number} priority
+   */
+  registerAnchorMutator(callback, priority) {
+    const am = new AnchorMutator(callback, priority);
+    if (this.registeredAnchorMutators_.length) {
+      this.registeredAnchorMutators_.splice(
+          this.findIndexToInsert(this.registeredAnchorMutators_, am), 0, am);
```

Comment:
A simpler option is to use a sparse array and simply do:

```js
user().assert(!this.anchorMutators[priority]);
this.anchorMutators[priority] = callback;
```

Then just add a null check when you iterate over `this.anchorMutators_`.

Reply:
This is definitely simpler and as discussed offline, I am not familiar with the performance of sparse arrays. Lets do this solution and follow up with research on what would be more performant. it also looks like forEach skips the empty cells so null check is not necessary.